### ISSUE TITLE
Notifications in different languages

### ISSUE THREAD

**Issue by saschabrockel** [louislam/uptime-kuma#456]
**Describe the solution you'd like**
It would be nice if the notifications would be in the same language as the interface or that you could choose the language or even individualize them. Currently, I've set the web interface to German but I'm still getting my mails, Discord & Telegram notifications in English. The problem is that not everybody speaks English especially older people.

**Comment by zsxeee** [comment]
Seems we need the server-side translation, including response message and notification message.

**Comment by Saibamen** [comment]
@zsxeee The simplest way -> return message codes (with parameters if any) from server. Frontend will translate it with interpolation (https://www.i18next.com/translation-function/interpolation)

**Comment by Saibamen** [comment]
But for notifications, IMO it will be better if translation keys will be in new object-key, like:

```
notifications: {
   LOGIN_SUCCESS: "Logged in successfully",
   UNEXPECTED_ERROR: "Unexpected error detected. Please try again, check if you have latest app version or report a bug in GitHub"
}
```

**Comment by zsxeee** [comment]
@Saibamen After check "notification-providers", I found many notification messages have their message text with little difference. It will make many unnecessary translation works.
Before add translation keys, I think we need a common message format/template first.

**Comment by louislam** [comment]
> @Saibamen After check "notification-providers", I found many notification messages have their message text with little difference. It will make many unnecessary translation works.
> Before add translation keys, I think we need a common message format/template first.

Agree, but it is not an easy task, as some notifications' payload is very special such as Discord and MS Teams.

**Comment by CommanderStorm** [comment]
I don't think that we will add this.
This is a fairly large issue and i18n in our backend is a larger task as-is.

If someone is willing to work on this, I think time is much better spend on the following issue instead
- https://github.com/louislam/uptime-kuma/issues/646

TL;DR: Some monitors already are templatable (!= translated), but the list is small.

If someone wants to do an RFC what it would take to move towards this goal, I wold like to read is.
Otherwise, this issue is not planned
